io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -289,18 +289,184 @@ impl Lvol {
             .expect("Receiver is gone");
     }
 
+    /// Get a `PtplFileOps` from `&self`.
+    pub(crate) fn ptpl(&self) -> impl PtplFileOps {
+        LvolPtpl::from(self)
+    }
+}
+
+struct LvolPtpl {
+    lvs: super::lvs_store::LvsPtpl,
+    uuid: String,
+}
+impl LvolPtpl {
+    fn lvs(&self) -> &super::lvs_store::LvsPtpl {
+        &self.lvs
+    }
+    fn uuid(&self) -> &str {
+        &self.uuid
+    }
+}
+impl From<&Lvol> for LvolPtpl {
+    fn from(lvol: &Lvol) -> Self {
+        Self {
+            lvs: (&lvol.lvs()).into(),
+            uuid: lvol.uuid(),
+        }
+    }
+}
+impl PtplFileOps for LvolPtpl {
+    fn create(&self) -> Result<Option<PtplProps>, std::io::Error> {
+        if let Some(path) = self.path() {
+            self.lvs().create()?;
+            if let Some(parent) = path.parent() {
+                std::fs::create_dir_all(parent)?;
+            }
+            return Ok(Some(PtplProps::new(path)));
+        }
+        Ok(None)
+    }
+
+    fn destroy(&self) -> Result<(), std::io::Error> {
+        if let Some(path) = self.path() {
+            std::fs::remove_file(path)?;
+        }
+        Ok(())
+    }
+
+    fn subpath(&self) -> std::path::PathBuf {
+        self.lvs()
+            .subpath()
+            .join("replica/")
+            .join(self.uuid())
+            .with_extension("json")
+    }
+}
+
+#[async_trait(?Send)]
+pub trait LogicalVolume {
+    /// returns the name of the Logical Volume
+    fn name(&self) -> String;
+
+    /// returns the UUID of the Logical Volume
+    fn uuid(&self) -> String;
+
+    /// return lvs for the Logical Volume
+    fn lvs(&self) -> Lvs;
+
+    /// returns the pool name of the Logical Volume
+    fn pool_name(&self) -> String;
+
+    /// returns the pool uuid of the Logical Volume
+    fn pool_uuid(&self) -> String;
+
+    /// returns a boolean indicating if the Logical Volume is thin provisioned
+    fn is_thin(&self) -> bool;
+
+    /// returns a boolean indicating if the Logical Volume is read-only
+    fn is_read_only(&self) -> bool;
+
+    /// return the size of the Logical Volume in bytes
+    fn size(&self) -> u64;
+
+    /// Returns Lvol disk space usage.
+    fn usage(&self) -> LvolSpaceUsage;
+}
+
+#[async_trait(?Send)]
+pub trait SpdkLvol: LogicalVolume + Share {
```

Comment:
What's the difference between `SpdkLvol` and `LogicalVolume` ?

Reply:
There is a plan to implement a new test framwork from LogicalVolume trait. Hence certein fileds are exposed through LogicalVolume which will have different implementation in the test framwork.

Reply:
Handled as per the suggestion. Please check once.